src/ivq.tsx

```diff
@@ -1,30 +1,69 @@
 import {h} from 'preact';
-import { IvqConfig } from "./types/IvqConfig";
+import {QuizLoader} from './providers/quiz-loader';
+import {IvqConfig} from './types/IvqConfig';
 
-// @ts-ignore
 export class Ivq extends KalturaPlayer.core.BasePlugin {
   private _player: KalturaPlayerTypes.Player;
+  private _resolveReadyPromise = () => {};
+  private _readyPromise: Promise<void>;
 
-  static defaultConfig: IvqConfig = {
-
-  };
+  static defaultConfig: IvqConfig = {};
 
   constructor(name: string, player: any, config: IvqConfig) {
     super(name, player, config);
     this._player = player;
+    this._readyPromise = this._makeReadyPromise();
   }
 
   get ready() {
-    return Promise.resolve();
+    return this._readyPromise;
   }
 
-  loadMedia(): void {}
+  loadMedia(): void {
+    const kalturaCuePointService: any = this._player.getService('kalturaCuepoints');
+    this._getQuestionsAndAnswers(kalturaCuePointService);
```

Comment:
Please add some check the kalturaCuePointService exists and if not return before _getQuestionsAndAnswers an warn to console